add natural key example (#5350)
fishtown-analytics/docs.getdbt.com: website/docs/docs/build/entities.md

There are four entity types: primary, foreign, unique, or natural.

:::tip Use entities as dimensions

> You can also use entities as dimensions, which allows you to aggregate a metric to the granularity of that entity.

Reviewer: nice catch!

* **Primary &mdash;** A primary key has **only one** record for each row in the table, and it includes every record in the data platform.
* **Unique &mdash;** A unique key contains **only one** record per row in the table, but it may have a subset of records in the data warehouse. It can also include nulls.
* **Foreign &mdash;** A foreign key can include zero, one, or multiple instances of the same record. Null values may also be present.
* **Natural &mdash;** Natural keys are columns or combinations of columns in a table that uniquely identify a record based on real-world data. For instance, in a sales_person_department dimension table, the sales_person_id can serve as a natural key. You can only use natural keys for [SCD type II dimensions](/docs/build/dimensions#scd-type-ii).

The following is the complete spec for entities:

```yaml
entities:
  - name: transaction     ## Required
    type: Primary or natural or foreign or unique ## Required
    description: A description of the field or role the entity takes in this table ## Optional
    expr: The field that denotes that entity (transaction_id).  ## Optional

>           If not specified, it will default to name 

Reviewer: ```suggestion
          Defaults to name if unspecified.
```

    expr: substring(id_order from 2)
```

### Combine columns with a key

> If there's a scenario where a table doesn't have a primary key, use the 'surrogate combination' to combine two columns as a key, helping you uniquely identify a record. This applies to any [entity type](/docs//build/entities#entity-types). For example, you can combine `date_key` and `brand_code` from the `raw_brand_target_weekly` table to form a `surrogate key`. The following example creates a `surrogate key` by joining `date_key` and `brand_code` using a pipe (`|`) as a separator. 

Reviewer 2: It doesn't have to be "primary"! could be any key

Reviewer 3: In the verbal example, it talks about not having a primary key, so I think for that example, it should be primary key. Here is what I ended up using for my most recent build:
```
# -----------------------
# --- semantic model ---
  - name: sem_department_target_weekly
    model: ref('fact_department_target_weekly')
    description: Describes weekly sales target for department for the current year
    defaults:
      agg_time_dimension: date_field

# --- entities ---
    entities:
      - name: sem_department_target_weekly_pk
        type: primary
        expr: date_key || '|' || department_code
      - name: date_key
        type: foreign
      - name: department_code
        type: foreign
```

Author: thanks @rpourzand and @patkearns10 - i've taken both of your feedback and changed it to the below. let me know what you think:

> If there's a scenario where a table doesn't have any key, such as a primary key, use the 'surrogate combination' to combine two columns as a key to help you identify a record. This applies to any [entity type](/docs//build/entities#entity-types)...

> If there's a scenario where a table doesn't have any key, such as a primary key, use the 'surrogate combination' to combine two columns as a key to help you identify a record. This applies to any [entity type](/docs//build/entities#entity-types). For example, you can combine `date_key` and `brand_code` from the `raw_brand_target_weekly` table to form a `surrogate key`. The following example creates a `surrogate key` by joining `date_key` and `brand_code` using a pipe (`|`) as a separator. 

Reviewer: sugg: use italics for emphasis 

```suggestion
If a table doesn't have any key (like a primary key), use _surrogate combination_ to form a key that will help you identify a record by combining two columns. This applies to any [entity type](/docs//build/entities#entity-types). For example, you can combine `date_key` and `brand_code` from the `raw_brand_target_weekly` table to form a _surrogate key_. The following example creates a surrogate key by joining `date_key` and `brand_code` using a pipe (`|`) as a separator. 
```
